src/app.service.ts

```diff
@@ -43,25 +52,61 @@ export class AppService {
     return this.gamesRepository.findOne(id);
   }
 
-  async createNewGame(size: number): Promise<Game> {
+  async createNewGame(size: number, player1SessionId: string): Promise<Game> {
     if (!size) {
       size = DEFAULT_BOARD_SIZE;
     } else {
       size = +size;
     }
-    const game = await this.gamesRepository.save(
-      this.gamesRepository.create({
-        state: initNewGameState(size),
-      }),
-    );
-    return game;
+    const player1 = this.usersRepository.create({
+      lastSessionId: player1SessionId,
+      username: player1SessionId
+    });
+    const game = this.gamesRepository.create({
+      state: initNewGameState(size),
+      player1: await this.usersRepository.save(player1),
+    });
+    return await this.gamesRepository.save(game);
   }
 
-  async createNewGameFromFile(): Promise<Game> {
-    let game = this.gamesRepository.create({
+  async createNewGameFromFile(player1SessionId: string): Promise<Game> {
+    const player1 = this.usersRepository.create({
+      lastSessionId: player1SessionId,
+      username: player1SessionId
+    });
+    const game = this.gamesRepository.create({
       state: this.getBoardStateFromFile(),
+      player1: await this.usersRepository.save(player1),
     });
-    game = await this.gamesRepository.save(game);
-    return game;
+    return await this.gamesRepository.save(game);
+  }
+
+  async joinGame(gameId: number, player2SessionId: string): Promise<Game> {
+    const player2 = this.usersRepository.create({
```

Comment:
you shouldn't crerate a player before checking that the game doesn't already have 2